src/main/java/euteam/cleanmachine/model/facility/Machine.java

```diff
@@ -1,83 +1,138 @@
 package euteam.cleanmachine.model.facility;
 
-import euteam.cleanmachine.model.facility.machine.state.IdleState;
-import euteam.cleanmachine.model.facility.machine.state.MachineState;
+import euteam.cleanmachine.exceptions.StateTransitionException;
+import euteam.cleanmachine.model.facility.machine.state.*;
+import org.hibernate.annotations.GenericGenerator;
 
 import javax.persistence.*;
 import java.util.List;
 
 @Entity
 public abstract class Machine {
     @Id
-    @GeneratedValue(strategy = GenerationType.AUTO)
-    private Long id;
+    @GeneratedValue(generator = "uuid")
+    @GenericGenerator(name = "uuid", strategy = "uuid")
+    private String identifier;
```

Comment:
I thought of the identifier as the machine serial number, the one provided by the manufacturer

Reply:
Oke keep it like this, except we don't generate the id but give it in the constructor? I assume serial number is unique?